xdsl/ir.py

```diff
@@ -231,13 +235,12 @@ def owner(self) -> Block:
         return self.block
 
     def __repr__(self) -> str:
-        if isinstance(self.block, Block):
-            block_repr = f"Block(num_arguments={len(self.block.args)}, " + \
-                         f"num_blocks={len(self.block.ops)} ops)"
-        else:
-            block_repr = repr(self.block)
-        return f"OpResult(typ={repr(self.typ)}, num_uses={repr(len(self.uses))}" + \
-               f", block={block_repr}, index={repr(self.index)}"
+        return "{}<%{}: {}>(num_uses={})".format(
+            self.__class__.__name__,
+            self.index,
```

Comment:
I think Block Arguments can have names too, no? Could we try to use it here too?

Reply:
Oh yes good point, I missed that!